File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/models/RoutineStep.kt
package com.HammersTech.RoutineChart.core.domain.models

import java.time.Instant

/**
 * RoutineStep domain model
 * Represents a single step within a routine
 * Steps are queried by routineId only, so they don't need userId or familyId
 */
data class RoutineStep(
    val id: String,
    val routineId: String,
    val orderIndex: Int,
    val label: String?,
    val iconName: String?,
    val audioCueUrl: String?,
    val createdAt: Instant,
    val deletedAt: Instant?
) {
    val isDeleted: Boolean
        get() = deletedAt != null
}

File: ios/RoutineChart/RoutineChart/Core/Domain/Models/RoutineStep.swift
//
//  RoutineStep.swift
//  RoutineChart
//
//  Domain model for RoutineStep entity
//

import Foundation

struct RoutineStep: Identifiable, Codable, Equatable, Hashable, Sendable {
    let id: String
    let routineId: String
    var orderIndex: Int
    var label: String?
    var iconName: String?
    var audioCueUrl: String?
    let createdAt: Date
    var deletedAt: Date?
    
    var isDeleted: Bool {
        deletedAt != nil
    }
    
    init(
        id: String = UUID().uuidString,
        routineId: String,
        orderIndex: Int,
        label: String? = nil,
        iconName: String? = nil,
        audioCueUrl: String? = nil,
        createdAt: Date = Date(),
        deletedAt: Date? = nil
    ) {
        self.id = id
        self.routineId = routineId
        self.orderIndex = orderIndex
        self.label = label
        self.iconName = iconName
        self.audioCueUrl = audioCueUrl
        self.createdAt = createdAt
        self.deletedAt = deletedAt
    }
}

